packages/gitbook/src/app/(site)/(content)/[[...pathname]]/page.tsx

```diff
@@ -168,27 +169,38 @@ export async function generateMetadata({
 /**
  * Fetches the page data matching the requested pathname and fallback to root page when page is not found.
  */
-async function getPageDataWithFallback(args: {
-    pagePathParams: PagePathParams;
-    searchParams: { fallback?: string };
-    redirectOnFallback?: boolean;
```

Comment:
Not sure why it as `true` for `Page` and false for `generatePageMetadata`, I don't see a reason for that. We can bring it back of course if it makes sense.

Reply:
I remember there was a reason, but it should have been documented with a comment...

Let's add it back instead of taking risks, or you need to try the different cases.

maybe @taranvohra @spastorelli remember?